pkg/apis/core/v1alpha2/types.go

```diff
@@ -129,6 +132,11 @@ type GroupSpec struct {
 	// Cannot be set with any other selector except NamespaceSelector.
 	// +optional
 	ExternalEntitySelector *metav1.LabelSelector `json:"externalEntitySelector,omitempty"`
+	// Select other ClusterGroups by name. The ClusterGroups must already
+	// exist and must not contain ChildGroups themselves.
```

Comment:
Question - if I apply a single yaml that creates a group and its child groups together, could the creation fail due to ordering?

Reply:
Good point, the order do matter in this case, but this brings up another problem: if child group comes before the parent group, then create can succeed but deletion with the same yaml would fail... I remember @antoninbas has the same concerns regarding this kind of constraints, since it exists in ACNP today as well (a CG can only be referenced in ACNP if it exists). Maybe we should considering lifting it for nested groups?

The 'gotcha' I can think of is, for example, if we have CG1 which has CG2 as child, and CG2 which has CG3 as child. It used to be we can only create CG3, CG2, CG1 in that order and CG3 creation will fail. Now, CG3 can be created first, and CG2 creation will fail. I suppose that is acceptable....

Reply:
Yeah, I always feel we should allow dangling references. Do not mean we must fix it in your PR.

Reply:
If everyone agrees and does not see any risks with dangling reference, I can definitely remove the constraint in this PR. @antoninbas @tnqn

Reply:
How about let us do that together with NetworkPolicy, once we agree that is the right approach to go.

Reply:
works for me